crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -859,7 +956,7 @@ impl<'a> IRGenerator<'a> {
         Ok(exit_first) //TODO what should we return???
     }
 
-    pub fn acir(&self, evaluator: &mut Evaluator) {
+    pub fn acir(&mut self, evaluator: &mut Evaluator) {
```

Comment:
It does not need to, good catch. It is a left-over because I need to modify the memory object hold by self inside the acir process. However Rust ownership did not allow me to so I ended up storing the info I need in the acir object.